### PR TITLE
use new package versions

### LINKED ISSUE #475: Update interfaces to match SF v0.8.0
```[tasklist]
### Tasks
- [x] Update `SurfaceValues`, `InteriorValues` > `StateValues`
- [x] Remove redundant aerodynamic roughness coefficients if necessary. Check that `PointValueScheme()` is being used.
- [x] Up ClimaLSM.jl: https://github.com/CliMA/ClimaLSM.jl/pull/362
```
cc @juliasloan25 @LenkaNovak

### PR DESCRIPTION
**Pull request by Sbozzolo**
- ClimaAtmos v0.16.3 (cache)
- SF v0.8 (interface)
  - #480 changes included here
- ClimaLSM v0.3.4

closes #475 and #490

replaces #500, #498, #473

### CODE REVIEW

File: experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/dummy_surface_fluxes.jl
Comment (akshaysridhar) [inline experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/dummy_surface_fluxes.jl:98]: Before we merge this can we see if this #TODO block can be resolved ?

File: experiments/AMIP/modular/components/land/bucket_init.jl
Comment (akshaysridhar) [inline experiments/AMIP/modular/components/land/bucket_init.jl:193]: I suppose this is a consequence of ClimaLSM updates ?

**Review by akshaysridhar — COMMENTED** [review]
Results from buildkite outcomes in the Slack thread LGTM (the run has been re-triggered, I'll review again once it is complete). Can we see if we can resolve the pending `TODO` statement with the updated package version?

File: Project.toml
Comment (LenkaNovak) [inline Project.toml:44]: Glad this is now working! 👍

File: Project.toml
Comment (LenkaNovak) [inline Project.toml:30]: Is this patch spec necessary?

File: Project.toml
Comment (LenkaNovak) [inline Project.toml:38]: Is this patch version still necessary?

**Review by LenkaNovak — APPROVED** [review]
LGTM after resolving @akshaysridhar 's comments (no major change in output diagnostics or performance, though please double check all runs once `build_history` is generated). Thank you all!

File: experiments/AMIP/modular/components/land/bucket_init.jl
Comment (LenkaNovak) [inline experiments/AMIP/modular/components/land/bucket_init.jl:192]: This could be causing the timeout issue in the distributed CI run (I saw similar errors in the long runs). It might be worth checking that `comms_ctx`  is being passed to ClimaLSM properly.

File: experiments/AMIP/modular/components/land/bucket_init.jl
Comment (juliasloan25) [reply experiments/AMIP/modular/components/land/bucket_init.jl:193]: Yes! ClimaLSM should have had a minor release bc of this interface change

File: Project.toml
Comment (juliasloan25) [reply Project.toml:30]: Yes, to help with some of the MPI/HDF5 problems that came up last week

Comment (Sbozzolo) [reply Project.toml:30]: It should be `0.16.3`. The code won't work with any previous version of `ClimaAtmos`.

Comment (LenkaNovak) [reply Project.toml:30]: I meant the the ClimaComms dep :)

Comment (juliasloan25) [reply Project.toml:30]: Ahh good point, I do need to specify the ClimaAtmos patch (oops)

File: Project.toml
Comment (juliasloan25) [reply Project.toml:38]: I'm not sure. Do you know why it was added in #482 ?

File: experiments/AMIP/modular/components/land/bucket_init.jl
Comment (juliasloan25) [reply experiments/AMIP/modular/components/land/bucket_init.jl:192]: ClimaLSM did switch to retrieving the comms_ctx from the space, under the assumption that the simulation and space contexts would be the same. It looks like in the coupler we're getting the space from `CA.get_integrator(atmos_config)` created [here](https://github.com/CliMA/ClimaCoupler.jl/blob/main/experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl#L67), then the space is accessed [here](https://github.com/CliMA/ClimaCoupler.jl/blob/main/experiments/AMIP/modular/coupler_driver_modular.jl#L215) and used throughout the simulation. Is that right? It seems to me that we would want the simulation to use the same context as the space it's running on

Comment (LenkaNovak) [reply experiments/AMIP/modular/components/land/bucket_init.jl:192]: For the current distributed runs, we do want to share the MPI context (and the `boundary_space`) across the models. But in the construction of the LSM column space [here](https://github.com/CliMA/ClimaCoupler.jl/blob/main/experiments/AMIP/modular/components/land/bucket_utils.jl#L31), `comms_ctx` doesn't seem to be communicated (and it is this column space that seems to be used to define MPI context within ClimaLSM). @akshaysridhar is seeing the same timeouts with the new ClimaLSM, though he's currently double checking it's coming from the albedo line. Perhaps we can meet offline once we know this for sure.

File: Project.toml
Comment (LenkaNovak) [reply Project.toml:39]: The new version was breaking the HDF5 dep. We can keep it for now, given that Atmos is having related issues and we are stable.

File: experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/dummy_surface_fluxes.jl
Comment (juliasloan25) [reply experiments/ClimaCore/tc4_atm-lnd-sfc-fluxes/dummy_surface_fluxes.jl:98]: we're planning on deleting this test case in the next couple of weeks, so I think it's okay to leave the TODO in this PR

**Comment by sriharshakandala** [comment]
Can we remove `UnPack` package? Its functionality can be replaced by destructuring syntax supported in native Julia!

**Comment by juliasloan25** [comment]
> Can we remove `UnPack` package? Its functionality can be replaced by destructuring syntax supported in native Julia!

I want to get this PR merged today, so I'll remove `UnPack` separately. I opened an issue for it here https://github.com/CliMA/ClimaCoupler.jl/issues/520

**Comment by juliasloan25** [comment]
Plots do look slightly different, but I think it's within acceptable bounds given there are no overall trend changes and we've updated for a new SurfaceFluxes in this PR. I'll merge now

**Pull request merged**
